spek-ide-plugin/intellij-common/src/main/kotlin/org/spekframework/intellij/base.kt

```diff
@@ -0,0 +1,97 @@
+package org.spekframework.intellij
+
+import com.intellij.execution.CommonProgramRunConfigurationParameters
+import com.intellij.execution.configuration.EnvironmentVariablesComponent
+import com.intellij.execution.configurations.ConfigurationFactory
+import com.intellij.execution.configurations.ConfigurationTypeBase
+import com.intellij.execution.configurations.ModuleBasedConfiguration
+import com.intellij.execution.configurations.RunConfigurationModule
+import com.intellij.openapi.util.JDOMExternalizerUtil
+import org.jdom.Element
+import org.jetbrains.kotlin.idea.KotlinIcons
+import org.spekframework.spek2.runtime.scope.Path
+import org.spekframework.spek2.runtime.scope.PathBuilder
+import org.spekframework.spek2.runtime.scope.isRoot
+
+abstract class SpekBaseConfigurationType(id: String, displayName: String): ConfigurationTypeBase(
+    id,
+    displayName,
+    "Run specifications",
+    KotlinIcons.SMALL_LOGO_13
+)
+
+
+abstract class SpekBaseRunConfiguration<T: RunConfigurationModule>(name: String,
+                                                                   configurationModule: T,
+                                                                   factory: ConfigurationFactory)
+    : ModuleBasedConfiguration<T>(name, configurationModule, factory), CommonProgramRunConfigurationParameters {
+
+    private var workingDirectory: String? = null
+    private var envs = mutableMapOf<String, String>()
+    private var passParentEnvs: Boolean = false
+    private var programParameters: String? = null
```

Comment:
This whole section of `var` seems like a potential multithreading hell, even if it is not quite possible. What do you think about `Atomic*` everything?

Reply:
ack

Reply:
A reminder in case you forgot this.

Reply:
I'll address this in a separate PR 👍